contracts/overrides/MultiReceiverRoyaltyOverrideCore.sol

```diff
@@ -0,0 +1,140 @@
+import "./IRoyaltySplitter.sol";
+import "../specs/IEIP2981.sol";
+
+/**
+ * Multi-receiver EIP2981 reference override implementation
+ */
+abstract contract EIP2981MultiReceiverRoyaltyMultiReceiverOverrideCore is IEIP2981, IEIP2981MultiReceiverRoyaltyOverride, ERC165 {
+    uint16 private _defaultRoyaltyBPS;
+    address payable private _defaultRoyaltySplitter;
+
+    mapping(uint256 => address payable) private _tokenRoyaltiesSplitter;
+    mapping(uint256 => uint16) private _tokenRoyaltiesBPS;
+    uint256[] private _tokensWithRoyalties;
+    
+    // Address of cloneable splitter contract
+    address internal _royaltySplitterCloneable;
+
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165, IERC165) returns (bool) {
+        return interfaceId == type(IEIP2981).interfaceId || interfaceId == type(IEIP2981MultiReceiverRoyaltyOverride).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @dev Sets token royalties. When you override this in the implementation contract
+     * ensure that you access restrict it to the contract owner or admin
+     */
+    function _setTokenRoyalties(TokenRoyaltyConfig[] memory royaltyConfigs) internal {
+        for (uint i; i < royaltyConfigs.length; i++) {
+            TokenRoyaltyConfig memory royaltyConfig = royaltyConfigs[i];
+            require(royaltyConfig.royaltyBPS < 10000, "Invalid bps");
+            Recipient[] memory recipients = royaltyConfig.recipients;
+            address payable splitterAddress = _tokenRoyaltiesSplitter[royaltyConfig.tokenId];
+            if (recipients.length == 0) {
+                if (splitterAddress != address(0)) {
+                    IRoyaltySplitter(splitterAddress).setRecipients(recipients);
+                }
+                delete _tokenRoyaltiesBPS[royaltyConfig.tokenId];
+                emit TokenRoyaltyRemoved(royaltyConfig.tokenId);
+            } else {
+                if (splitterAddress == address(0)) {
+                    splitterAddress = payable(Clones.clone(_royaltySplitterCloneable));
+                    RoyaltySplitter(splitterAddress).initialize(recipients);
+                    _tokenRoyaltiesSplitter[royaltyConfig.tokenId] = splitterAddress;
+                    _tokensWithRoyalties.push(royaltyConfig.tokenId);
+                } else {
+                    IRoyaltySplitter(splitterAddress).setRecipients(recipients);
+                }
+                _tokenRoyaltiesBPS[royaltyConfig.tokenId] = royaltyConfig.royaltyBPS;
+                emit TokenRoyaltySet(royaltyConfig.tokenId, royaltyConfig.royaltyBPS, recipients);
+            }
+        }
+    }
+
+    /**
+     * @dev Sets default royalty. When you override this in the implementation contract
+     * ensure that you access restrict it to the contract owner or admin
+     */
+    function _setDefaultRoyalty(uint16 bps, Recipient[] memory recipients) internal {
+        require(bps < 10000, "Invalid bps");
```

Comment:
probably shouldnt have this check in the contract but on client-side if needed as in our case